packages/api-format/src/codec/base/Option.ts

```diff
@@ -0,0 +1,63 @@
+// Copyright 2017-2018 @polkadot/api-format authors & contributors
+// This software may be modified and distributed under the terms
+// of the ISC license. See the LICENSE file for details.
+
+import { Base } from '../types';
+
+export default class CodecOption implements Base<Base<any> | null> {
```

Comment:
Based on the length of the first byte, it determines if something follows or not. If 0, nothing follows, if 1, we have actual data f a specific type.